server/boutique-hotel-technikum/src/main/java/com/example/boutiquehoteltechnikum/services/BookingService.java

```diff
@@ -0,0 +1,76 @@
+package com.example.boutiquehoteltechnikum.services;
+
+import com.example.boutiquehoteltechnikum.models.BookingEntity;
+import com.example.boutiquehoteltechnikum.models.GuestEntity;
+import com.example.boutiquehoteltechnikum.models.RoomEntity;
+import com.example.boutiquehoteltechnikum.objects.BookingRequestObject;
+import com.example.boutiquehoteltechnikum.repositories.BookingRepository;
+import com.example.boutiquehoteltechnikum.repositories.GuestRepository;
+import com.example.boutiquehoteltechnikum.utils.DateValidator;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.List;
+
+@Service
+public class BookingService {
+
+	@Autowired
+	private BookingRepository bookingRepository;
+	@Autowired
+	private GuestRepository guestRepository;
+	@Autowired
+	private RoomService roomService;
+
+	public BookingEntity addBooking(BookingRequestObject bookingRequestObject) {
+		RoomEntity room = roomService.getRoomById(
+			bookingRequestObject.getRoomId(),
+			DateValidator.getDate(bookingRequestObject.getStartDate()),
+			DateValidator.getDate(bookingRequestObject.getEndDate())
+		);
+		if (room == null)
+		{
+			throw new ResponseStatusException(
+				HttpStatus.NOT_FOUND, "Das Zimmer ist zu den angegebenen Daten nicht mehr verfügbar!"
+			);
+		}
+
+		GuestEntity guestEntity = guestRepository.save(GuestEntity.builder()
```

Comment:
// ISSUE: Users with the same email address cannot make multiple bookings?

Reply:
As the email is not a unique key, this should not be a problem